src/tribler-core/tribler_core/modules/tunnel/tests/test_triblertunnel_community.py

```diff
@@ -627,3 +629,35 @@ async def test_perform_http_request_failed(self):
             await wait_for(self.nodes[0].overlay.perform_http_request(('127.0.0.1', 1234),
                                                                       b'GET /scrape?info_hash=0 HTTP/1.1\r\n\r\n'),
                            timeout=.3)
+
+    def test_exitnode_settings(self):
```

Comment:
You could make this test more concise using the already available `TestBase` tools:

<details>
  <summary>Click me for suggestion.</summary>

```patch
diff --git a/src/tribler-core/tribler_core/modules/tunnel/tests/test_triblertunnel_community.py b/src/tribler-core/tribler_core/modules/tunnel/tests/test_triblertunnel_community.py
index 817dfc461..f955b99e5 100644
--- a/src/tribler-core/tribler_core/modules/tunnel/tests/test_triblertunnel_community.py
+++ b/src/tribler-core/tribler_core/modules/tunnel/tests/test_triblertunnel_community.py
@@ -42,11 +42,14 @@ class TestTriblerTunnelCommunity(TestBase):  # pylint: disable=too-many-public-m
             await node.overlay.bandwidth_community.unload()
         await super().tearDown()
 
-    def create_node(self):
-        mock_ipv8 = MockIPv8("curve25519", TriblerTunnelCommunity,
+    def create_node(self, *args, **kwargs):
+        mock_ipv8 = MockIPv8("curve25519",
+                             TriblerTunnelCommunity,
                              socks_listen_ports=[],
-                             settings={'remove_tunnel_delay': 0},
-                             exitnode_cache=mkdtemp(suffix="_tribler_test_cache") / 'cache.dat')
+                             exitnode_cache=mkdtemp(suffix="_tribler_test_cache") / 'cache.dat',
+                             *args,
+                             **kwargs)
+        mock_ipv8.overlay.remove_tunnel_delay = 0
         mock_ipv8.overlay.settings.max_circuits = 1
 
         # Load the bandwidth accounting community
@@ -641,23 +644,8 @@ class TestTriblerTunnelCommunity(TestBase):  # pylint: disable=too-many-public-m
         session.config.get_state_dir = lambda: Path('STATE_DIR')
         session.config.get_tunnel_community_socks5_listen_ports = lambda: [1234]
 
-        settings = {'foo': 'bar'}
+        self.initialize(TriblerTunnelCommunity, 1, tribler_session=session)
 
-        my_peer = Mock()
-        my_peer.key = LibNaCLPK(pk='a' * 32, hex_vk='b' * 32)
-
-        endpoint = Mock()
-        endpoint.is_open = lambda: False
-
-        network = Mock()
-
-        community = TriblerTunnelCommunity(my_peer, endpoint, network,
-                                           tribler_session=session, settings=settings)
-        try:
-            assert PEER_FLAG_EXIT_BT in community.settings.peer_flags
-            assert PEER_FLAG_EXIT_IPV8 in community.settings.peer_flags
-            assert PEER_FLAG_EXIT_HTTP in community.settings.peer_flags
-        finally:
-            community.cancel_all_pending_tasks()
-            community.request_cache.cancel_all_pending_tasks()
-            community.dispatcher.cancel_all_pending_tasks()
+        assert PEER_FLAG_EXIT_BT in self.overlay(0).settings.peer_flags
+        assert PEER_FLAG_EXIT_IPV8 in self.overlay(0).settings.peer_flags
+        assert PEER_FLAG_EXIT_HTTP in self.overlay(0).settings.peer_flags
```

</details>